src/main/webapp/app/overview/course-exams/course-exam-detail/course-exam-detail.component.ts

```diff
@@ -93,32 +97,37 @@ export class CourseExamDetailComponent implements OnInit, OnDestroy {
             this.cancelExamStateSubscription();
             return;
         }
-        if (dayjs(this.exam.startDate).isAfter(dayjs())) {
-            if (dayjs(this.exam.startDate).diff(dayjs(), `s`) < 600) {
+        if (dayjs().isBefore(this.exam.startDate)) {
+            if (dayjs(this.exam.startDate).diff(dayjs(), `seconds`) < 600) {
                 this.examState = ExamState.IMMINENT;
             } else {
                 this.examState = ExamState.UPCOMING;
             }
             this.timeLeftToStartInSeconds();
             return;
         }
-        if (
-            this.exam.examStudentReviewStart &&
-            this.exam.examStudentReviewEnd &&
-            dayjs().isBetween(dayjs(this.exam.examStudentReviewStart), dayjs(this.exam.examStudentReviewEnd))
-        ) {
-            this.examState = ExamState.STUDENTREVIEW;
+        if (dayjs().isBefore(this.exam.endDate)) {
+            this.examState = ExamState.CONDUCTING;
             return;
         }
-        if (dayjs(this.exam.endDate).isAfter(dayjs())) {
-            this.examState = ExamState.CONDUCTING;
+
+        this.loadStudentExam();
+        if ((!this.studentExam || !this.studentExam.submitted) && !this.exam.testExam) {
```

Comment:
Something is not working right here. I never get into the STUDENTREVIEW ExamState even though I am currently in between examStudentReviewStart and examStudentReviewEnd. The card just always says "Exam Closed". I guess it has something to do with this if clause. Also maybe some more comments detailing what is happening here?

Reply:
We already have a comment right below: `// Exam is over and student did not participate. We can cancel the subscription`, i.e. the student did not successfully participate in the normal exam. I moved the comment one line up

Reply:
I mean a more general explanation, i.e. why we have to load the student exam here if the method name indicates just checking the current exam state and what the different checks are for. It is not very obvious why there are two different checks that might lead to `ExamState.CLOSED`. Just my two cents from trying to understand this code without having worked with it before :)